cudax/include/cuda/experimental/__device/arch_traits.cuh

```diff
@@ -0,0 +1,401 @@
+
+//! @brief Retrieve architecture traits of the specified architecture
+//!
+//! @param __sm_version Compute capability in 10 * major + minor format for which the architecture traits are requested
+//!
+//! @throws cuda_error if the requested architecture is unknown
+_CCCL_HOST_DEVICE inline constexpr arch_traits_t arch_traits(unsigned int __sm_version)
+{
+  switch (__sm_version)
+  {
+    case 60:
+      return detail::sm_600_traits;
+    case 70:
+      return detail::sm_700_traits;
+    case 75:
+      return detail::sm_750_traits;
+    case 80:
+      return detail::sm_800_traits;
+    case 86:
+      return detail::sm_860_traits;
+    case 90:
+      return detail::sm_900_traits;
+    default:
+      __throw_cuda_error(cudaErrorInvalidValue, "Traits requested for an unknown architecture");
+      break;
+  }
+}
+
+//! @brief Type representing a CUDA device architecture. It provides traits from arch_traits_t in form of static members
+template <unsigned int __SmVersion>
+struct arch : public detail::arch_common_traits
+{
+private:
+  static constexpr arch_traits_t __traits = detail::arch_traits<__SmVersion>();
+
+public:
+  static constexpr int compute_capability_major             = __traits.compute_capability_major;
+  static constexpr int compute_capability_minor             = __traits.compute_capability_minor;
+  static constexpr int compute_capability                   = __traits.compute_capability;
+  static constexpr int max_shared_memory_per_multiprocessor = __traits.max_shared_memory_per_multiprocessor;
+  static constexpr int max_warps_per_multiprocessor         = __traits.max_warps_per_multiprocessor;
+  static constexpr int max_blocks_per_multiprocessor        = __traits.max_blocks_per_multiprocessor;
+  static constexpr int max_threads_per_multiprocessor       = __traits.max_threads_per_multiprocessor;
+  static constexpr int reserved_shared_memory_per_block     = __traits.reserved_shared_memory_per_block;
+  static constexpr int max_shared_memory_per_block_optin    = __traits.max_shared_memory_per_block_optin;
+
+  static constexpr bool cluster_supported  = __traits.cluster_supported;
+  static constexpr bool redux_intrinisic   = __traits.redux_intrinisic;
+  static constexpr bool elect_intrinsic    = __traits.elect_intrinsic;
+  static constexpr bool cp_async_supported = __traits.cp_async_supported;
+  static constexpr bool tma_supported      = __traits.tma_supported;
+
+  constexpr operator arch_traits_t() const
+  {
+    return __traits;
+  }
+};
+
+// TODO: possible this should be __device__ constexpr variable?
+//! @brief Provides architecture traits of the architecture matching __CUDA_ARCH__ macro
+_CCCL_DEVICE constexpr inline arch_traits_t current_arch()
+{
+#ifdef __CUDA_ARCH__
+  return detail::arch_traits<__CUDA_ARCH__ / 10>();
+#else
+  // Should be unreachable in __device__ function
+  return arch_traits_t{};
+#endif
+}
+
+// TODO might need to remove this since sm_80 and sm_86 is both called ampere :(
```

Comment:
maybe `ampere_sm80`, `ampere_sm86`? all the others have a name, e.g. Orin, Ada, etc.

Reply:
I ended up removing these, we can add them later if needed. You can instead do arch<800>, arch<860> etc